sdk/tracing/src/test/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessorTest.java

```diff
@@ -171,25 +174,35 @@ void exportMoreSpansThanTheBufferSize() {
     ReadableSpan span4 = createSampledEndedSpan(SPAN_NAME_1);
     ReadableSpan span5 = createSampledEndedSpan(SPAN_NAME_1);
     ReadableSpan span6 = createSampledEndedSpan(SPAN_NAME_1);
-    List<SpanData> exported = waitingSpanExporter.waitForExport();
-    assertThat(exported)
-        .containsExactly(
-            span1.toSpanData(),
-            span2.toSpanData(),
-            span3.toSpanData(),
-            span4.toSpanData(),
-            span5.toSpanData(),
-            span6.toSpanData());
+
+    // Give time for the BatchSpanProcessor to attempt to export spans.
+    Thread.sleep(500);
```

Comment:
Using `Thread.sleep` is fragile in tests

Reply:
I agree - I couldn't find a way around it for this test though.

Reply:
> I agree - I couldn't find a way around it for this test though.

I had the same problem in my PR. The test should assert only what it can guarantee ie that it’ll flush the max buffer and nothing else. If we aren’t happy with that then we should remove the test entirely, as the required outcome can’t be achieved. 

Flaky tests will cause headaches for some unfortunate soul down the track.